src/Microsoft.AspNet.OData.Shared/EdmChangedObjectCollection.cs

```diff
@@ -57,5 +69,288 @@ private void Initialize(IEdmEntityType entityType)
+                            if (status == PatchStatus.Success)
+                            {
+                                IEdmChangedObject changedObject = HandleFailedOperation(deletedObj, operation, original, keys, errorMessage);
+                                changedObjectCollection.Add(changedObject);
+                                continue;
+                            }
+                        }
+
+                        changedObjectCollection.Add(deletedObj);
+                    }
+                    else
+                    {
+                        EdmEntityObject deltaEntityObject = changedObj as EdmEntityObject;
+
+                        PatchStatus status = PatchHandler.TryGet(keyValues, out original, out geterrorMessage);
+
+                        if (status == PatchStatus.NotFound)
+                        {
+                            operation = DataModificationOperationKind.Insert;
+
+                            if (PatchHandler.TryCreate(changedObj, out original, out errorMessage) != PatchStatus.Success)
+                            {
+                                //Handle failed Opreataion - create
+                                IEdmChangedObject changedObject = HandleFailedOperation(deltaEntityObject, operation, original, keys, errorMessage);
+                                changedObjectCollection.Add(changedObject);
+                                continue;
+                            }
+                        }
+                        else if (status == PatchStatus.Success)
+                        {
+                            operation = DataModificationOperationKind.Update;
+                        }
+                        else
+                        {
+                            //Handle failed operation 
+                            IEdmChangedObject changedObject = HandleFailedOperation(deltaEntityObject, operation, null, keys, geterrorMessage);
+                            changedObjectCollection.Add(changedObject);
+                            continue;
+                        }                                              
+
+                        //Patch for addition/update. 
+                        PatchItem(deltaEntityObject, original);
+
+                        changedObjectCollection.Add(changedObj);
+                    }
+                }
+                catch(Exception ex)
+                {
+                    //Handle Failed Operation
+                    IEdmChangedObject changedObject = HandleFailedOperation(changedObj as EdmEntityObject, operation, originalObj, keys, ex.Message);
+                    
+                    Contract.Assert(changedObject != null);
+                    changedObjectCollection.Add(changedObject);
+                }
+            }
+
+            return changedObjectCollection;
+        }
+
+    
+        private static IDictionary<string, object> GetKeyValues(List<IEdmStructuralProperty> keys, IEdmChangedObject changedObj)
```

Comment:
>GetKeyValues [](start = 51, length = 12)

What is the payload uses @odata.id, rather than key values? Do we not support that? Or do we try and handle that when we deserialize the resource?

Reply:
We have planned to use it as back up right? in case id is not present